Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -3,6 +3,274 @@ import mongoc
+            case let .success(events):
+                var seenEvents: [ChangeStreamTestEvent] = []
+                for _ in 0..<events.count {
+                    let event = try changeStream.nextOrError()
+                    expect(event).toNot(beNil())
+                    seenEvents.append(event!)
+                }
+                expect(seenEvents).to(equal(events), description: self.description)
+            }
+        } catch {
+            self.result.matchesError(error: error, description: self.description)
+        }
+    }
+}
+
+/// Struct representing a single change-streams spec test JSON file.
+private struct ChangeStreamTestFile: Decodable {
+    private enum CodingKeys: String, CodingKey {
+        case databaseName = "database_name",
+             collectionName = "collection_name",
+             database2Name = "database2_name",
+             collection2Name = "collection2_name",
+             tests
+    }
+
+    /// Name of this test case
+    var name: String = ""
+
+    /// The default database.
+    let databaseName: String
+
+    /// The default collection.
+    let collectionName: String
+
+    /// Secondary database
+    let database2Name: String
+
+    // Secondary collection
+    let collection2Name: String
+
+    /// An array of tests that are to be run independently of each other.
+    let tests: [ChangeStreamTest]
+}
+
+final class ChangeStreamSpecTests: MongoSwiftTestCase, FailPointConfigured {
+    var activeFailPoint: FailPoint?
+
+    override func tearDown() {
+        self.disableActiveFailPoint()
+    }
+
+    func testChangeStreamSpec() throws {
+        let testFilesPath = MongoSwiftTestCase.specsPath + "/change-streams/tests"
+        let testFiles: [String] = try FileManager.default.contentsOfDirectory(atPath: testFilesPath)
+
+        let tests: [ChangeStreamTestFile] = try testFiles.map { fileName in
+            let url = URL(fileURLWithPath: "\(testFilesPath)/\(fileName)")
+            var testFile = try BSONDecoder().decode(ChangeStreamTestFile.self, from: Document(fromJSONFile: url))
+
+            testFile.name = fileName
+            return testFile
+        }
+
+        let globalClient = try MongoClient(MongoSwiftTestCase.connStr)
+
+        let version = try globalClient.serverVersion()
+        let topology = MongoSwiftTestCase.topologyType
+
+        for testFile in tests {
+            let db1 = globalClient.db(testFile.databaseName)
+            let db2 = globalClient.db(testFile.database2Name)
+            print("\n------------\nExecuting tests from file \(testFilesPath)/\(testFile.name)...\n")
+            for test in testFile.tests {
+                try db1.drop()
+                try db2.drop()
+                _ = try db1.createCollection(testFile.collectionName)
+                _ = try db2.createCollection(testFile.collection2Name)
+
+                let testTopologies = test.topology.map { TopologyDescription.TopologyType(from: $0) }
```

Comment:
should we do the topology/version checks before we bother dropping DBs and creating collections?